### PR TITLE
Add Documentation for Install Knative with Ambassador

### PR DESCRIPTION
**Pull request by nbkrause**
Ambassador recently released an update that gives it the ability to create routes based off of Knative `ClusterIngress` resources. 

This PR adds Ambassador to the list of envoy-based solutions for install Knative.

### CODE REVIEW

**Comment by samodell** [comment]
Hey @nbkrause -- can you or @richarddli verify that he has signed a Google CLA?

**Comment by richarddli** [comment]
Google has a corporate CLA for Datawire (covers both me and @nbkrause).

**Comment by richarddli** [comment]
@samodell Is there anything else you need in order to merge? (We have quite a number of users who are asking for Knative docs, so it would be helpful to have this merged so we can point people to it.)

File: docs/install/Knative-with-GKE.md [hunk 1]
```diff
@@ -118,8 +118,8 @@ the recommended configuration for a cluster is:
 1. Create a Kubernetes cluster on GKE with the required specifications:
 
 > Note: If this setup is for development, or a non-Istio networking layer (e.g.
-> [Gloo](./Knative-with-Gloo.md)) will be used, then you can remove the
-> `--addons` line below.
+> [Ambassador](https://getambassador.io/) or [Gloo](./Knative-with-Gloo.md))
```
Comment (samodell) [inline docs/install/Knative-with-GKE.md]: I think this link to Ambassador should point to the page in Knative Docs, instead of to the Ambassador site.

**Comment by samodell** [comment]
@richarddli  This looks generally good! I had one suggestion, but otherwise we can merge this in.

File: docs/install/Knative-with-GKE.md (hunk 1 above)
Comment (nbkrause) [reply docs/install/Knative-with-GKE.md]: @samodell Good find. Thank you for taking the time to review my work and find that error.

**Comment by nbkrause** [comment]
@samodell Should I perform the cherry-pick against the release-0.7 branch to push these docs live?

**Comment by samodell** [comment]
@nbkrause  Yes, that would be great!

**Pull request merged**